src/index.js

```diff
@@ -96,24 +98,133 @@ export async function getArgs () {
       console.log(result)
     })
 
+  program
+    .command('installAppBundle <InstalledAppId> <AgentHash> <AppBundleSource>')
+    .alias('b')
+    .option('-e --membraneProof <membraneProof>', 'provide membrane proof for app bundle install and runtime validation - should be paired with --cellNick')
+    .option('-n --cellNick <cellNick>', 'provide cell nick - should be paired with --membraneProof')
+    .option('-u --uid <uid>', 'provide uid to happ')
+    .description('install provided happ bundle with given id and details: calls InstallAppBundle(installed_app_id, agent_key, source, membrane_proofs?, uid?) -> { installed_app_id, cell_data: [ { cell_id, cell_nick } ], status: { inactive: { reason: [Object] } } }')
+    .action(async (InstalledAppId, AgentHash, AppBundleSource, { membraneProof, cellNick, uid }) => {
+      let membraneProofs
+      // throw error if one, but now both, of the membraneProof and CellNick are present - both are needed to form the membrane_proofs object
+      if (!membraneProof ^ !cellNick) {
+        throw new Error('When installing a happ with membrane proof, both the --membraneProof and --cellNick options are required.')
+      } else if (membraneProof) {
+        membraneProofs = {
+          [[cellNick]]: Buffer.from(membraneProof, 'base64')
+        }
+      }
+
+      let bundleSource
+      // eslint-disable-next-line no-useless-escape
+      const urlMatcher = /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)?/gi
+      if (AppBundleSource.match(urlMatcher)) {
+        if (path.extname(AppBundleSource) !== '.happ') {
+          throw new Error('Invalid AppBundleSource URL.  Please ensure to provide the url address of your bundle\'s .happ file.')
+        }
+        console.log('Downloading bundle URL...', AppBundleSource)
+        bundleSource = await downloadFile(AppBundleSource)
+      } else {
+        try {
+          const isHappExt = path.extname(AppBundleSource) === '.happ'
+          if (!isHappExt) throw new Error('Invalid AppBundleSource file path.  Please ensure to provide the url address of your bundle\'s .happ file.')
+        } catch (error) {
+          throw new Error('Invalid AppBundleSource. The source must be either a file or url path to the .happ file.')
+        }
+        // note: __dirname is configured to the 'dist' folder, not project root
+        const bundlePath = path.join(__dirname, '../', AppBundleSource)
```

Comment:
What is the reasoning behind this?

Reply:
That is bc in the webpack config, the root source is set to the dist folder, so in order to get the path from the project directory for the app path, we will always need to go out one.